### ISSUE TITLE
Google/(FaceBook?) Ads Are Not Being Blocked Anymore

### ISSUE THREAD

**Issue by w4tadrag** [blokadaorg/blokada#91]
Update 1: here is a temporary fix for ads that are appearing on chrome: https://www.xda-developers.com/fix-dns-ad-blocker-chrome/
But this doesn't prevent ads from appearing on apps.

Update 2: I checked CCleaner again and it looks like the ads are provided by Facebook. If that's the case, then ads from FB are not being blocked. Also when I go to websites (i.e. news website) from the Facebook app with the built-in browser, I see ads that seem like are from Facebook.
----
Hi, recently I have noticed that google ads are not being blocked in many apps as well as on the browsers such as chrome. I have not changed anything in the settings. I also tried reinstalling, restarting my phone, still didn't work. I have attached some pictures that show the ads that I am seeing. 

Device: Oneplus One
OS version: 5.1.1. 

Edit: added device info.

![image](https://user-images.githubusercontent.com/32439406/35230223-7abea07a-ff63-11e7-8be4-593754247c5b.png)

![image](https://user-images.githubusercontent.com/32439406/35230238-836c7f26-ff63-11e7-894a-a3de8caf8d64.png)

![image](https://user-images.githubusercontent.com/32439406/35230254-8c269c00-ff63-11e7-8b55-0e5d240b21ec.png)

**Comment by kar** [comment]
Could you check your hosts count and blacklist, see if they are not empty by any chance?

**Comment by w4tadrag** [comment]
Hi, thank you for the reply. There is one in the blacklist. There are 42,934 hosts. Also, only "Unified hosts" file is on. 

Btw this is not only happening to your adblock, it's happening to many others.
Here is a temporary fix only for chrome that was posted yesterday on xda: https://www.xda-developers.com/fix-dns-ad-blocker-chrome/

But this doesn't prevent ads from appearing on apps.

**Comment by alleluia20** [comment]
I am now using DNS66 (without any apps in the whitelist, except for gmail), which is also DNS-based, and I am not finding this problem. It is open source, so you may want to have a look and see if you can find a solution.
By the way, I wonder whether this bug is related with #81, which I submitted some weeks ago.

**Comment by cam-mcnaught** [comment]
(at least when installed them) By default, Blokada only enables one blacklist while DNS66 enables 6. I enabled all the installed blacklist sources except HPHosts, and now I don't have problems with ads in apps again. Note that the first thing I did was disable the chrome flag as per the temporary fix, but haven't re-enabled it to see if it is still needed.

**Comment by w4tadrag** [comment]
Hi guys, so I did a little more testing with few more apps. So when I have Blokada enabled and tried running Speed Test app, I don't see any ads. I tried the same for CCleaner, I see ads. Here is a crazy assumption, what's the chance CCleaner is using the same method as the YouTube app to provide the ads as they cannot be blocked? 

There is another adblocking app I used in the past that requires more permissions to monitor encrypted data to block the ads, but the downside was that the company wants to track what the users do and sell that data, thus I don't use it anymore.

Btw, I have tried using DNS66 in the past and even today. It didn't block any ads in the past for me, neither did it now.

**Comment by alleluia20** [comment]
w4tadrag, this is probably because DNS66 whitelists almost all system apps, which causes that most in-app ads show. But you can remove all the apps from the whitelist at once, and then whitelist the apps you wish. Then, if you find any problems with system upgrades, you can temporarily disable the adblocker.

**Comment by w4tadrag** [comment]
@kar I checked CCleaner again and it looks like the ads are provided by Facebook I think. So when I press the AdChoice (blue arrow looking thingy at the bottom left) it takes me to FB. If that's the case, then ads from FB are not being blocked. 

Here is what happens when I press AdChoice:
![image](https://user-images.githubusercontent.com/32439406/35703965-e4126808-076b-11e8-9111-5e4367778496.png)

Not sure if you want me to create a new issue for it.

**Comment by rpeter85** [comment]
To test if it solves the issue, can you guys try to add the below blocklist in Blokada and check if you still see ads on/from facebook?
https://raw.githubusercontent.com/StevenBlack/hosts/master/extensions/social/hosts

**Comment by rpeter85** [comment]
@wrtadrag yes, that list blocks _all_ social sites, unfortunately. What you can do is to copy the facebook links from that, create a local file on your phone with those links and use the file in Blokada.

**Comment by w4tadrag** [comment]
@rpeter85 oh yeah true, I could do that. Thanks :)

**Comment by rpeter85** [comment]
@w4tadrag I took a look on the sponsored content on FB. As I can see and understand, they are "embedded", means not the full link is present. So, either all the fb-links are blocked (just like the 'social' hosts list does) or have to figure out which fb-server/s provide/s the ads.

**Comment by kar** [comment]
This is a great discussion guys, very helpful. I've added the social extension list (disabled by default), it'll be available in the next release, and after it's done I'll backport it to older releases too.

I don't think we can do much more on this issue as long as we are doing dns-based blocking, and more advanced blocking is a long way to go (but I hope we'll get there). Another useful thing would be to introduce better UI for whitelisting / blacklisting hosts on the fly, so that it'd be easy to create your own list. Feel free to create another issue for this feature.

I'm closing this one, but more input is always welcome :)

**Comment by w4tadrag** [comment]
@kar just keep in mind that this list will block all the social media websites. I have attached an updated version which removed the links that would block the websites and **was still able to block ads** that were provided by Facebook in apps or website (except video ads for some reason). I did not try to check pinterest or myspace as I don't use them. 
[social media adblock list.txt](https://github.com/blokadaorg/blokada/files/1740483/social.media.adblock.list.txt)


Thanks again for your help @rpeter85. Yeah they are embedded on FB, Instagram, and Twitter.
